libcudacxx/test/libcudacxx/std/language.support/cstdlib/abs.pass.cpp

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/std/cassert>
+#include <cuda/std/cstdlib>
+#include <cuda/std/limits>
+
+#include "test_macros.h"
+
+using I   = int;
+using IL  = long;
+using ILL = long long;
+
+__host__ __device__ constexpr I abs_overload(I value)
+{
+  ASSERT_SAME_TYPE(I, decltype(cuda::std::abs(I{})));
+  static_assert(noexcept(cuda::std::abs(I{})), "");
+  return cuda::std::abs(value);
+}
+
+__host__ __device__ constexpr IL abs_overload(IL value)
+{
+  ASSERT_SAME_TYPE(IL, decltype(cuda::std::labs(IL{})));
+  static_assert(noexcept(cuda::std::labs(IL{})), "");
+  return cuda::std::labs(value);
+}
+
+__host__ __device__ constexpr ILL abs_overload(ILL value)
+{
+  ASSERT_SAME_TYPE(ILL, decltype(cuda::std::llabs(ILL{})));
+  static_assert(noexcept(cuda::std::llabs(ILL{})), "");
+  return cuda::std::llabs(value);
+}
+
+template <class T>
+__host__ __device__ TEST_CONSTEXPR_CXX14 bool test_abs(T in, T ref, T zero_value)
+{
+  assert(abs_overload(zero_value + in) == ref);
```

Comment:
Q: why is adding `zero_value` needed? Why not:
```suggestion
  assert(abs_overload(in) == ref);
```

Reply:
And adding a zero will prevent constant folding of integers? :D I can only trust you on this one.

Reply:
It is not adding a zero, it is adding a value loaded from a pointer